ES1 Katas - Learn by Re-writing - array.indexOf

shinenelson opened the pull request.

shinenelson committed:
learn ES1 array.indexOf by re-writing

sirajulmcb commented on learn-by-rewriting/es1/array-api/indexOf.js:
```diff
@@ -26,11 +26,23 @@
+  if( !isFinite(fromIndex) ) {
+    fromIndex = 0
+  }
+  for ( let index = fromIndex; index <= array.length; index++  ) {
```
`index` would be `fromIndex` if positive
`index` would be `array.length + fromIndex` if negative. You can make index 0 if this offset is below zero. Not sure if that is necessery.

shinenelson replied:
I'm already handling the negative value in the `if` condition above just like you're mentioning here - i.e., set `fromIndex = 0`. Is that what you mean?

There were tests with negative `fromIndex` which forced me to write that check before I got to the `for` loop.

shinenelson replied:
I've added some checks for the negative `fromIndex` as per the [`indexOf` specification](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) per your suggestion

shinenelson committed:
[es1][learn by rewriting][array API] add check for negative fromIndex

resolves review https://github.com/shinenelson/learning-ecmascript/pull/26/files#r434853800 by @sirajulm
